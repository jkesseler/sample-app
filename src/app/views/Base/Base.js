import React from 'react';
import PropTypes from 'prop-types';
import CssBaseline from '@material-ui/core/CssBaseline';
import { Paper } from '@material-ui/core';

import Header from '../../components/Header';


const AppBase = ({ children }) => (
  <CssBaseline>
    <Paper
      elevation={0}
      style={{
        padding: 0,
        margin: 0,
        backgroundColor: '#fafafa',
      }}
    >
      <Header />
      { children }
    </Paper>
  </CssBaseline>
);

AppBase.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default AppBase;
