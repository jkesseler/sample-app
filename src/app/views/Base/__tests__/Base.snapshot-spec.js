import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Base from '../Base';

const Child = () => (<span>fake child</span>);

describe('Base', () => {
  it('renders without crashing', () => {
    const component = shallow(<Base><Child /></Base>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
