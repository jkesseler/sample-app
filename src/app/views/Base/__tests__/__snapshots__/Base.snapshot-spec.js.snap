// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders without crashing 1`] = `
<WithStyles(CssBaseline)>
  <WithStyles(Paper)
    elevation={0}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <React.Memo />
    <Child />
  </WithStyles(Paper)>
</WithStyles(CssBaseline)>
`;
